### PR TITLE
Update gradle to version 8.6

### PR DESCRIPTION
**Pull request by juliagarrigos**
Gradle updated to version `8.6`, it required:
- Updating gradle tools to version `8.2.2`
- Minor structure changes in the gradle files
- Moving the namespace to the gradle file
- Adding the property `useAndroidX=true` 

We also added a comment explaining the complexity of updating the `UiAutomator` library to the latest version.

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -7,8 +7,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.4.2'
-        
+        classpath 'com.android.tools.build:gradle:7.0.4'
```
Comment (MyDogTom) [inline server/build.gradle]: @juliagarrigos are there any reasons why you chose AGP 7.0.4 and Gradle 7.2 instead of picking latest versions?

**Review by MyDogTom — APPROVED** [review]
@juliagarrigos LGTM. Just a a question out of curiosity.

File: server/build.gradle (hunk 1 above)
Comment (juliagarrigos) [reply server/build.gradle]: I wanted to keep the changes in this PR limited, updating to gradle 8 was requiring extra changes that required extra investigation. I'll try to create a PR soon updating to the latest gradle version.

Comment (juliagarrigos) [reply server/build.gradle]: In the end the changes required were not that important so I managed to include the update in this PR 💃🏻

**Review by MyDogTom — APPROVED** [review]
👍

**Pull request merged**